ddtrace/internal/datadog/profiling/stack_v2/src/sampler.cpp

```diff
@@ -24,33 +24,29 @@ Sampler::sampling_thread()
             });
         });
 
-        // If we've been asked to stop, then stop
-        if (!is_profiling.load()) {
-            break;
-        }
-
         // Sleep for the remainder of the interval, get it atomically
         // Generally speaking system "sleep" times will wait _at least_ as long as the specified time, so
         // in actual fact the duration may be more than we indicated.  This tends to be more true on busy
         // systems.
         std::this_thread::sleep_until(sample_time_now + microseconds(sample_interval_us.load()));
     }
-
-    // Release the profiling mutex
-    profiling_mutex.unlock();
 }
 
 void
 Sampler::set_interval(double new_interval_s)
 {
-    unsigned int new_interval_us = static_cast<unsigned int>(new_interval_s * 1e6);
+    microsecond_t new_interval_us = static_cast<unsigned int>(new_interval_s * 1e6);
     sample_interval_us.store(new_interval_us);
 }
 
 void
 Sampler::set_max_nframes(unsigned int _max_nframes)
```

Comment:
Why is this not part of one-time-setup?

Reply:
Ah, this isn't straightforward.  I factored it out differently, which removes the propagation of max nframes from this context.